app/src/main/java/com/babylon/wallet/android/domain/usecases/transaction/PollTransactionStatusUseCase.kt

```diff
@@ -21,44 +25,47 @@ class PollTransactionStatusUseCase @Inject constructor(
         transactionType: TransactionType = TransactionType.Generic,
         endEpoch: Epoch
     ): TransactionStatusData {
-        val txId = intentHash.bech32EncodedTxId
+        return withContext(dispatcher) {
+            val sargonOs = sargonOsManager.sargonOs
+            val txId = intentHash.bech32EncodedTxId
 
-        return when (val transactionStatus = transactionRepository.pollTransactionStatus(intentHash)) {
-            is TransactionStatus.Failed -> TransactionStatusData(
-                txId = txId,
-                requestId = requestId,
-                result = Result.failure(
-                    if (transactionStatus.reason == TransactionStatusReason.WORKTOP_ERROR) {
-                        RadixWalletException.TransactionSubmitException.TransactionCommitted.AssertionFailed(txId)
-                    } else {
-                        RadixWalletException.TransactionSubmitException.TransactionCommitted.Failure(txId)
-                    }
-                ),
-                transactionType = transactionType
-            )
-            is TransactionStatus.PermanentlyRejected -> TransactionStatusData(
-                txId = txId,
-                requestId = requestId,
-                result = Result.failure(RadixWalletException.TransactionSubmitException.TransactionRejected.Permanently(txId)),
-                transactionType = transactionType
-            )
-            TransactionStatus.Success -> TransactionStatusData(
-                txId = txId,
-                requestId = requestId,
-                result = Result.success(Unit),
-                transactionType = transactionType
-            )
-            is TransactionStatus.TemporarilyRejected -> TransactionStatusData(
-                txId = txId,
-                requestId = requestId,
-                result = Result.failure(
-                    RadixWalletException.TransactionSubmitException.TransactionRejected.Temporary(
-                        txId = txId,
-                        txProcessingTime = (endEpoch - transactionStatus.currentEpoch).toMinutes().toString()
-                    )
-                ),
-                transactionType = transactionType
-            )
+            when (val transactionStatus = sargonOs.pollTransactionStatus(intentHash)) {
```

Comment:
In relation to the previous comment TransactionStatusData can be a sealed class too with different states instead of keeping a result inside. Only `TransactionStatus.Failed` keeps an error. If that looks like a huge task I would not consider fixing it now.